lib/fastlane/actions/update_url_schemes.rb

```diff
@@ -0,0 +1,68 @@
+require 'plist'
+
+module Fastlane
+  module Actions
+    class UpdateUrlSchemesAction < Action
+      def self.run(params)
+        path = params[:path]
+        url_schemes = params[:url_schemes]
+
+        hash = Plist.parse_xml(path)
+        hash['CFBundleURLTypes'].first['CFBundleURLSchemes'] = url_schemes
+        File.write(path, hash.to_plist)
+      end
+
+      def self.description
+        'Updates the URL schemes in the given Info.plist'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :path,
+            env_name: 'FL_UPDATE_URL_SCHEMES_PATH',
+            description: 'The Plist file\'s path',
+            is_string: true,
+            verify_block: verify_path
+          ),
+
+          FastlaneCore::ConfigItem.new(
+            key: :url_schemes,
+            env_name: "FL_UPDATE_URL_SCHEMES_URL_SCHEMES",
+            description: 'The new URL schemes',
+            is_string: false,
+            verify_block: verify_url_schemes
+          )
+        ]
+      end
+
+      def self.output
+        []
+      end
+
+      def self.authors
+        ['kmikael']
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac].include? platform
+      end
+
+      def self.verify_path
+        lambda do |v|
+          unless File.exist?(v)
+            raise "Could not find plist at path '#{v}'".red
+          end
+        end
+      end
+
+      def self.verify_url_schemes
+        lambda do |v|
+          unless v.kind_of?(Array) && v.inject(true) { |m, o| m && o.kind_of?(String) }
```

Comment:
It should be easier for non-Ruby developers to read the code. How about something like
```ruby
raise "url schemes must be an array".red unless v.kind_of?(Array)
v.each do |current|
  raise "Array must include strings only" unless current.kind_of?(String)
end
```